helm/cluster/templates/clusterapi/_helpers.tpl

```diff
@@ -51,7 +51,8 @@
 {{- end }}
 {{- /* Add provider-specific NO_PROXY values from template */}}
 {{- if $providerIntegration.connectivity.proxy.noProxy.templateName }}
-{{- range $noProxyAddress := include $providerIntegration.connectivity.proxy.noProxy.templateName $ | fromYamlArray }}
+{{- $values := dict "global" $global }}
+{{- range $noProxyAddress := include $providerIntegration.connectivity.proxy.noProxy.templateName (dict "Values" $values) | fromYamlArray }}
```

Comment:
Just for my understanding, the name "Values" in `dict "Values" $values` is arbitrary right? 

It could be also named "ProxyList" or anything else right?

Reply:
Actually no, it's to enable called template to use `.Values`.

e.g. here, template `"cluster.connectivity.proxy.noProxy"` calls another  template by name like this:

```
include $providerIntegration.connectivity.proxy.noProxy.templateName (dict "Values" $values)
```

e.g. for $providerIntegration.connectivity.proxy.noProxy.templateName value cluster-aws will pass `"awsNoProxyList"` template name, so we call template like that with that `dict` because `"awsNoProxyList"` template looks like this (it uses `$.Values`):

```
{{- define "awsNoProxyList" }}
- {{ $.Values.global.connectivity.network.vpcCidr }}
{{- end }}
```

So the caller of `"awsNoProxyList"` has to pass an argument (object/map) that has `.Values`

Reply:
Initially the call was made plainly like this:

```
include $providerIntegration.connectivity.proxy.noProxy.templateName $
```

so there was a Helm templating error like this:

```
executing "cluster.connectivity.proxy.noProxy" at <include $providerIntegration.connectivity.proxy.noProxy.templateName $>:

error calling include: template: cluster-aws/templates/_helpers.tpl:372:6:

executing "awsNoProxyList" at <$.Values.global.connectivity.network.vpcCidr>:

nil pointer evaluating interface {}.global
```